src/targets/gpu/hiprtc/main.cpp

```diff
@@ -31,12 +31,21 @@
 #include <iostream>
 #include <cstring>
 
+#ifdef _WIN32
+#include <fcntl.h>
+#include <io.h>
+#endif
+
 std::vector<char> read_stdin()
 {
     std::vector<char> result;
 
     std::array<char, 1024> buffer;
     std::size_t len = 0;
+#ifdef _WIN32
+    if(_setmode(_fileno(stdin), _O_BINARY) == -1)
```

Comment:
Can you add a comment here to explain the reason for this?

Reply:
Actually, maybe we should add a static method to do this so we dont need ifdefs here:

```cpp
struct process
{
    static void use_binary_for_stdin();
};
```